core/src/main/java/com/linecorp/armeria/common/util/Sampler.java

```diff
@@ -42,6 +42,64 @@
  */
 @FunctionalInterface
 public interface Sampler<T> {
+
+    /**
+     * Returns a sampler that applies logical or operator to both samplers decisions.
+     */
+    default Sampler<T> or(Sampler<T> other) {
+        return new OrSampler<>(this, other);
+    }
+
+    /**
+     * Returns a sampler that applies logical and operator to both samplers decisions.
+     */
+    default Sampler<T> and(Sampler<T> other) {
+        return new AndSampler<>(this, other);
+    }
+
+    /**
+     * Returns a sampler that returns {@code true} if the value is greater than the given value.
+     */
+    static <T extends Comparable<T>> Sampler<T> greaterThan(T val) {
+        return object -> object.compareTo(val) > 0;
```

Comment:
Comments on this method and below:
```suggestion
        requireNonNull(val, "val");
        return object -> object.compareTo(val) > 0;
```

Reply:
Good catch 👍